### DISCUSSION TITLE
Starting From Scratch...

### DISCUSSION THREAD

**Discussion by jrbgit** [massCodeIO/massCode#4] [General]
Congrats on taking the leap to start from scratch. It can be a hard decision to make sometimes.

Without going into too much detail - can you explain what made you decide to start from scratch? Maybe also some details about what the limitations were in the v1 branch and what you hope to overcome in v2?

Moving the repo to be under an organization instead of a personal account is a great start!

I starred the repo - and will be watching to see where this goes.

Cheers!

**Comment by antonreshetov** [comment]
The v1 code base is simply obsolete. massCode v1 was written using Electron v7, Vue 2, and JavaScript. The new version of massCode is based on Electron 16, Vue 3, and TypeScript. The only option to use the latest technology is to start from scratch, since rewriting the current code base is pretty much like starting from scratch, except that a lot of time will be spent migrating to the new requirements of the technology in use. This also includes updating dependencies. 

Electron v16 makes it possible to create a build for M1, Vue 3 & Vite is a very fast development and final build of the frontend of the application, TypeScript is a typing tool which gives less time for the subsequent development. 

In the new version of the application I am abandoning Monaco editor in favor of Ace editor, as it is a lighter weight library. Ace also supports much more language syntax highlighting and TextMate, which is more common than the proprietary Monarch for Monaco editor.

The new version will have a database based on plain JSON. In v1 it was used for this purpose library NeDB, which is no longer supported by the author and has bugs. Also, new version will have API server available on localhost, which will allow integration with any third-party applications, such as Raycast or Alfred.
